Dockerfile

```diff
@@ -15,19 +26,18 @@ EXPOSE 5050
 VOLUME /conf
 VOLUME /certs
 
-# Copy appdaemon into image
 WORKDIR /usr/src/app
-COPY . .
 
-# Install timezone data
-RUN apk add tzdata
+# Install runtime required packages
+# First line is required, 2nd line is for backwards compatibility
+RUN apk add --no-cache curl python3 py3-pip tzdata \
+        git py3-wheel build-base gcc libffi-dev openssl-dev musl-dev cargo
 
-# Install dependencies
-RUN apk add --no-cache build-base gcc libffi-dev openssl-dev musl-dev cargo rust curl \
-    && pip install --no-cache-dir .
+# Copy compiled deps from builder image
+COPY --from=builder /usr/lib/python3.10/site-packages/ /usr/lib/python3.10/site-packages/
 
-# Install additional packages
-RUN apk add --no-cache curl
+# Copy appdaemon into image
+COPY . .
```

Comment:
We should copy only the files related to the application, instead of the whole repository. That way we have a better chance of re-using the Docker layer cache, without introducing unrelated changed files in the image.